### PR TITLE
DisallowedEmptyRuleFixerRector: fix test expectations

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

File: rules-tests/Strict/Rector/Empty_/DisallowedEmptyRuleFixerRector/Fixture/empty_string_nullable.php.inc [hunk 1]
```diff
@@ -20,7 +20,7 @@ final class EmptyStringNullable
 {
     public function run(string|null $value)
     {
-        return $value === null || $value === '';
+        return $value === null || $value === '' || $value === '0';
```
Comment (staabm) [inline rules-tests/Strict/Rector/Empty_/DisallowedEmptyRuleFixerRector/Fixture/empty_string_nullable.php.inc:23]: `'0'` is also a empty string, therefore this test was wrong

see https://3v4l.org/mgSdd

Comment (samsonasik) [reply rules-tests/Strict/Rector/Empty_/DisallowedEmptyRuleFixerRector/Fixture/empty_string_nullable.php.inc:23]: the behavior probably based on configured `TREAT_AS_NON_EMPTY`

https://github.com/rectorphp/rector-src/blob/ecc0a2889157f7da9de1adee1a3f3f581844d2e2/rules/Strict/Rector/Empty_/DisallowedEmptyRuleFixerRector.php#L62

Comment (samsonasik) [reply rules-tests/Strict/Rector/Empty_/DisallowedEmptyRuleFixerRector/Fixture/empty_string_nullable.php.inc:23]: set `TREAT_AS_NON_EMPTY` to true seems make only compare null https://getrector.com/demo/204c8e52-11f0-492a-91a0-6b9f05c33531

**Comment by samsonasik** [comment]
@staabm I cherry-picked your commit at PR https://github.com/rectorphp/rector-src/pull/5296

**Comment by samsonasik** [comment]
Resolved at https://github.com/rectorphp/rector-src/pull/5296

**Comment by staabm** [comment]
Thanks 🙏

**Pull request closed without merging**